fix disable airspeed check with negative ASPD_FS_INTEG

ThomasRigi opened the pull request:
@sfuhrer  fix for https://github.com/PX4/PX4-Autopilot/issues/18185

sfuhrer commented on src/modules/airspeed_selector/AirspeedValidator.cpp:
```diff
@@ -149,6 +149,11 @@ AirspeedValidator::check_airspeed_innovation(uint64_t time_now, float estimator_
 		_innovations_check_failed = false;
 		_time_last_tas_pass = time_now;
 
+	} else if (_tas_innov_integ_threshold <= 0.f) {
```
Agree, the current logic is suboptimal in case of either bad mag/velocity data or a disabled integrator.

How about combining that condition (_tas_innov_integ_threshold <= 0.f) with the previous one? (if (!_in_fixed_wing_flight ..))

sfuhrer reviewed (commented):
While we're on it, shouldn't the check also be fully disabled if 
estimator_status_vel_test_ratio < 1.f) && (estimator_status_mag_test_ratio < 1.f)?

ThomasRigi commented:
To be honest I didn't fully understand how the check on the innovations was supposed to work, so I didn't touch it. 

If it's a test "if it doesn't pass then don't invalidate" then yes we can also put it with an OR statement in L148 together with !_in_fixed_wing_flight ..) and  (_tas_innov_integ_threshold <= 0.f)

sfuhrer commented:
> To be honest I didn't fully understand how the check on the innovations was supposed to work, so I didn't touch it.
> 
> If it's a test "if it doesn't pass then don't invalidate" then yes we can also put it with an OR statement in L148 together with !_in_fixed_wing_flight ..) and (_tas_innov_integ_threshold <= 0.f)

Yes I guess that was the original intent, run the check if the mag and vel test pass, and mark it pass otherwise. Even better would be to in that case only make it possible to set _innovations_check_failed to true, but not to false again (meaning that if you had it fail before, then you have the other estimation issues, it doesn't get reset).

ThomasRigi commented:
I pushed a new commit: 
- I addressed your comment on putting (_tas_innov_integ_threshold <= 0.f) in the first OR statement
- I think I implemented the second check as you said. If the estimator statuses are not good then don't reset `_innovations_check_failed` in case it was already in case it had already failed before when the estimates were still good, but still reset `_time_last_pass` to not potentially falsely invalidate the sensor immediately the first time the check is run again when the estimator statuses are validated again.

Can you please check if my implementation corresponds to what you had in mind?

sfuhrer commented on src/modules/airspeed_selector/AirspeedValidator.cpp:
```diff
@@ -145,34 +145,33 @@ AirspeedValidator::check_airspeed_innovation(uint64_t time_now, float estimator_
-			}
+		} else {
+			// reset integrator used to trigger and record pass if integrator check is disabled
+			_apsd_innov_integ_state = 0.f;
```
We should also reset the integrator state above, whenever we don't use it.

ThomasRigi replied:
Good suggestion, included in my new commit.

sfuhrer reviewed (commented):
The logic looks correct to me.

ThomasRigi commented:
@sfuhrer thanks for reviewing. I pushed a commit including your suggestions. Is it good to go in now?
